simpeg/objective_function.py

```diff
@@ -435,57 +447,61 @@ def multipliers(self):
         list of int
             Multipliers for the objective functions.
         """
-        return self._multipliers
+        return [obj.multiplier for obj in self.objfcts]
 
     @multipliers.setter
-    def multipliers(self, value):
+    def multipliers(self, value: list[float] | None):
         """Set multipliers attribute after checking if they are valid."""
+        if value is None:
+            return
+
         for multiplier in value:
             _validate_multiplier(multiplier)
+
         _check_length_objective_funcs_multipliers(self.objfcts, value)
-        self._multipliers = value
+
+        for val, obj in zip(value, self.objfcts):
+            obj.multiplier = val
```

Comment:
We can also keep the top-level `multipliers` assignment - they just get sent down the chain to their respective functions.